src/python/ksc/cgen.py

```diff
@@ -386,23 +386,23 @@ def concat_args(callable):
     // Make the first call to determine output size
     auto ret0 = ks::{ks_name}(&g_alloc {concat_args(lambda k: f", ks_arg{k}[0]")});
 
-    // Create Torch return value
+    // Create empty Torch return value
     auto [{ks_sizes}] = ret0.size();
-    // TODO: use torch::empty here, and copydown below.
-    auto ret = torch::zeros({{n, {ks_sizes}}});
+    auto ret = torch::empty({{n, {ks_sizes}}});
+    
     // And wrap it in ks - this is a view of the torch data, so convert_argument, not convert_return_value
     auto ks_ret = convert_argument<ks::tensor<{ks_return_dim}, Float>>(ret);
 
     // Place 0th value in the output
     auto ks_ret0 = ks_ret[0];
-    inplace_add(&ks_ret0, ret0); // This would update a temporary in the 1D case
+    inplace_copy(&ks_ret0, ret0); // This would update a temporary in the 1D case
```

Comment:
I'm confused here (not really by this PR, but by #1019). Is the comment saying that the 1D case doesn't work? I don't see how it would work.

Reply:
Yes, I'll clarify the comment.

Reply:
I've been wondering about making the 1D case more symmetric, e.g. Returning a dim0 tensor.

Reply:
Does it work to change the line above to `auto && ks_ret0 = ks_ret[0]` (and similarly inside the loop below).

Reply:
So IIUC, you're saying this:
```c++
  float a;
  float& f() { return a; }
  ks::tensor<float> b;
  ks::tensor<float> g() { return b; }

  auto& pa = f(); *(&pa) = 4; // changes a
  auto& pb = g(); *(&pb) = k; // can't assign rvalue to a reference

  auto&& pa = f(); *(&pa) = 4; // changes a
  auto&& pb = g(); *(&pb) = k; // changes b
```
?

Reply:
Thanks!